### PR TITLE
Fix non-blocking bugs

### PR DESCRIPTION
**Pull request by ceane**
Fixing various minor bugs within the app.

* Fix "X" close for modals
* Fix login wrapper in the menu
* Style guide requires login
* Transaction "See All" link doesn't work
* "Use Full Balance" link does nothing
* Validation on Preferences page
* Balance is not displaying correctly

### CODE REVIEW

**Comment by stubelshe** [comment]
@datchley what's the status of the review of this PR? It's become a blocker for a few final CSS changes.

File: src/components/Shortcuts/Shortcut.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const Shortcut = (item, _key) => {
   if (Array.isArray(item)) {
     return (
       <li key={_key}>
-        <Shortcuts data={item} />
+        <Shortcuts className="-is-logged-in" data={item} />
```
Comment (stubelshe) [inline src/components/Shortcuts/Shortcut.js:12]: @ceanela will this add -is-logged-in to the ul list container or the li's themselves?

File: src/components/Shortcuts/__snapshots__/test.js.snap [hunk 2]
```diff
@@ -6,7 +6,7 @@ exports[`<Shortcuts /> renders without crashing 1`] = `
 >
   <li>
     <ul
-      className="menulist"
+      className="menulist -is-logged-in"
```
Comment (stubelshe) [inline src/components/Shortcuts/__snapshots__/test.js.snap:9]: @ceanela same question as above, this seems to add the class to the ul itself, that would be the preference. We would not need the class added to the li's, just the parent ul.

**Review by stubelshe — CHANGES_REQUESTED** [review]
Questions added.

File: src/components/Shortcuts/Shortcut.js (hunk 1 above)
Comment (ceane) [reply src/components/Shortcuts/Shortcut.js:12]: This just adds it to the container

File: src/components/Shortcuts/__snapshots__/test.js.snap (hunk 2 above)
Comment (ceane) [reply src/components/Shortcuts/__snapshots__/test.js.snap:9]: Ignore any code in the snapshots please

File: src/components/Shortcuts/Shortcut.js (hunk 1 above)
Comment (stubelshe) [reply src/components/Shortcuts/Shortcut.js:12]: Perfect

File: src/components/Shortcuts/__snapshots__/test.js.snap (hunk 2 above)
Comment (stubelshe) [reply src/components/Shortcuts/__snapshots__/test.js.snap:9]: Will do.

**Review by stubelshe — APPROVED** [review]

**Pull request merged**